ayat/authentication_routes.py

```diff
@@ -0,0 +1,211 @@
+        user.name = data['name']
+        user.email = data['email']
+        user.password = generate_password_hash(data['password'], method='sha256')
+        user.country_name = data['country_name']
+        user.profile_picture = data['profile_picture']
+        user.phone_number = data['phone_number']
+        user.birth_date = data['birth_date']
+
+    if current_user['type'] == 'staff':
+        user.type = data['type']
+
+    db.session.commit()
+    return jsonify({'message': 'The user has been promoted!'})
+
+	
+
+@app.route('/v1/users/<public_id>',methods=['DELETE'])
+@token_required
+def delete_user(current_user,public_id):
+    if (not current_user['type'] == 'staff') or (not current_user['publid_id'] == str(public_id)):
+        return jsonify({"error": "user is unauthorized"}), 403
+    user = User.query.filter_by(public_id=public_id).first()
+
+    if not user:
+        return jsonify({'message': 'No user found!'})
+    db.session.delete(user)
+    db.session.commit()
+
+    return jsonify({'message':'The user has been deleted!'})
+
+
+@app.route('/v1/users', methods=['POST'])
+def login_or_create():
+
+    data = request.get_json()
+
+    # login checking
+    if data['action'] == 'login':
+
+        user_email = data['email']
+        user_password = data['password']
+
+        user = User.query.filter_by(email=user_email).first()
+
+        if not user:
+            return jsonify({"error": "user is unauthorized"}), 403
+
+        if check_password_hash(user.password,user_password):
+            token= jwt.encode({'public_id': user.public_id},app.config['SECRET_KEY'])
+            return jsonify({
+
+                            'token' : token.decode('UTF-8'),
+                            'public_id' : user.public_id,
+
+                            'name' : user.name,
+                            'email' : user.email ,
+                            'country_name' : user.country_name ,
+                            'phone_number' : user.phone_number ,
+                            'profile_picture' : user.profile_picture,
+                            'birth_date' : user.birth_date ,
+                            'gender' : user.gender ,
+            })
+        
+
+        return jsonify({"error": "user is unauthorized"}), 403
+
+    # creating a new user
+
+    if data['action'] == 'register' :
+
+        # checking if user exists or not 
+        user_email = data['email']
+        user = User.query.filter_by(email=user_email).first()
+        if user is not None:
+            return jsonify({"status":  "Email already exists"}), 1
+
+
+        user_phone = data['phone']
+        user = User.query.filter_by(phone=user_phone).first()
+        if user is not None:
+            return jsonify({"status":  "Phone already exists"}), 2
+
+
+
+        hashed_password = generate_password_hash(data['password'], method='sha256')
+
+        new_user = User(
+                        user_id = data['user_id'], ## ??????????

```

Comment:
Nope, we should generate a random id for each unique user via `UUID`. check issue references.